### PR TITLE
Playlist Management

### PR DESCRIPTION
**Pull request by surbhi84**
This pr consists of playlist management, adding removing videos from playlists and adding removing playlists.
added responsiveness in the PR as well.

https://tapesvideolibrary.netlify.app/
check this link deploy preview might not work.

### CODE REVIEW

File: src/backend/controllers/PlaylistController.js [hunk 1]
```diff
@@ -45,8 +45,11 @@ export const getAllPlaylistsHandler = function (schema, request) {
 export const addNewPlaylistHandler = function (schema, request) {
   const user = requiresAuth.call(this, request);
   if (user) {
-    const { playlist } = JSON.parse(request.requestBody);
-    user.playlists.push({ ...playlist, videos: [], id: uuid() });
+    const playlist = JSON.parse(request.requestBody);
+    const name = playlist.name;
+    const list = playlist.list === undefined ? [] : playlist.list;
+    console.log(playlist.list);
```
Comment (rohanmathur91) [inline src/backend/controllers/PlaylistController.js:51]: remove console log

File: src/backend/controllers/PlaylistController.js [hunk 2]
```diff
@@ -139,12 +136,14 @@ export const removeVideoFromPlaylistHandler = function (schema, request) {
...
     const videoId = request.params.videoId;
+    console.log(videoId);
     let playlist = user.playlists.find((item) => item.id === playlistId);
-    const filteredVideos = playlist.videos.filter(
-      (item) => item.id !== videoId
-    );
-    playlist.videos = filteredVideos;
+    console.log(playlist);
+    const filteredVideos = playlist.list.filter((item) => item.id !== videoId);
+    console.log(filteredVideos);
```
Comment (rohanmathur91) [inline src/backend/controllers/PlaylistController.js:145]: same as above, remove console logs

File: src/hooks/reducer/userReducer/reducer.js [hunk 3]
```diff
@@ -83,6 +87,64 @@ export const useUserReducer = () => {
...
+          user: {
+            ...state.user,
+            playlists: [...state.user.playlists, payload],
+          },
+        };
+      }
+
+      case DELPLAYLIST: {
+        const newPlaylists = state.user.playlists.filter(
+          (i) => i.id !== payload
```
Comment (rohanmathur91) [inline src/hooks/reducer/userReducer/reducer.js:103]: suggestion: give some meaningful name to the parameter

**Review by rohanmathur91 — APPROVED** [review]
- Liked that you are using Tailwind for styling.
- Good use of absolute imports, as its avoiding the nested paths
- Liked that you are sharing the stateful logic using custom hooks

UI suggestion:
- When hovering over a hamburger menu or any video card  make cursor pointer

File: src/App.js [hunk 4]
```diff
@@ -21,9 +21,8 @@ function App() {
       <Navbar />
       <Login />
       <Signup />
-      {/* SUCCESS_TOAST ONLY APPEARS IF THERE'S A SUCCESS MESSAGE TO SHOW */}
+      {/* SUCCESS_TOAST ONLY APPEARS IF THERE'S A MESSAGE TO SHOW,ALSO SUCCESS TOAST SHOWS ERRORS AS WELL FOR NOW*/}
```
Comment (SJTGSHIVAM) [inline src/App.js:24]: Ahh! these comments are really helping me read.

File: src/backend/controllers/PlaylistController.js [hunk 5]
```diff
@@ -45,8 +45,11 @@ export const getAllPlaylistsHandler = function (schema, request) {
 export const addNewPlaylistHandler = function (schema, request) {
   const user = requiresAuth.call(this, request);
   if (user) {
-    const { playlist } = JSON.parse(request.requestBody);
-    user.playlists.push({ ...playlist, videos: [], id: uuid() });
+    const playlist = JSON.parse(request.requestBody);
+    const name = playlist.name;
+    const list = playlist.list === undefined ? [] : playlist.list;
```
Comment (SJTGSHIVAM) [inline src/backend/controllers/PlaylistController.js:50]: I get the logic why we are doing it. this is making this API call flexible which is really interesting. I am just wondering is there any different way we can write. Just curious about it. This looks cool as well.

File: src/pages/LikedVids.jsx [hunk 6]
```diff
@@ -13,11 +13,11 @@ export function LikedVids() {
...
+      <h2 className="text-sm xs:text-lg font-semibold text-gray-700 mt-4">
         Liked Videos{" "}
       </h2>
-      <hr className="my-1 color-slate-300" />
-      {LikedVids > 0 ? (
+      <hr className="my-1 color-slate-300 mb-2" />
+      {likes.length > 0 ? (
         <>
           {likes.map(
             ({ id, title, creator, views, uploadedOn, img, avatar, about }) => (
```
Comment (SJTGSHIVAM) [inline src/pages/LikedVids.jsx:23]: Prefer having multiple attributes on multiple lines this is just better linting and makes code more maintainable.

File: src/pages/SinglePlaylist.jsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+          <h2 className="text-sm xs:text-lg font-semibold text-gray-700 mt-4">
+            {currentPlaylist.name}
+          </h2>
+          <hr className="my-1 color-slate-300" />
+
+          <div className="flex flex-col">
+            {currentPlaylist?.list.length > 0 ? (
+              <>
+                {currentPlaylist.list.map(
+                  ({
```
Comment (SJTGSHIVAM) [inline src/pages/SinglePlaylist.jsx:63]: here you already did it great.

**Review by SJTGSHIVAM — APPROVED** [review]
Approved.
Have added some of the suggestions, the overall code looks clean.

**Pull request merged**